python/selfie-lib/tests/SourceFile_test.py

```diff
@@ -0,0 +1,175 @@
+    assert str(source_file.parse_to_be_like(1).arg) == "7"
+
+    source_file = SourceFile("UnderTest.py", "  .toBe(\n7)  ")
+    assert str(source_file.parse_to_be_like(1).function_call_plus_arg) == ".toBe(\n7)"
+    assert str(source_file.parse_to_be_like(1).arg) == "7"
+
+    source_file = SourceFile("UnderTest.py", "  .toBe(7\n)  ")
+    assert str(source_file.parse_to_be_like(1).function_call_plus_arg) == ".toBe(7\n)"
+    assert str(source_file.parse_to_be_like(1).arg) == "7"
+
+
+def test_single_line_string():
+    source_file = SourceFile("UnderTest.py", ".toBe('7')")
+    assert str(source_file.parse_to_be_like(1).function_call_plus_arg) == ".toBe('7')"
+    assert str(source_file.parse_to_be_like(1).arg) == "'7'"
+
+    source_file = SourceFile("UnderTest.py", ".toBe('')")
+    assert str(source_file.parse_to_be_like(1).function_call_plus_arg) == ".toBe('')"
+    assert str(source_file.parse_to_be_like(1).arg) == "''"
+
+    source_file = SourceFile("UnderTest.py", ".toBe( '' )")
+    assert str(source_file.parse_to_be_like(1).function_call_plus_arg) == ".toBe( '' )"
+    assert str(source_file.parse_to_be_like(1).arg) == "''"
+
+    source_file = SourceFile("UnderTest.py", ".toBe( \n '' \n )")
+    assert (
+        str(source_file.parse_to_be_like(1).function_call_plus_arg)
+        == ".toBe( \n '' \n )"
+    )
+    assert str(source_file.parse_to_be_like(1).arg) == "''"
+
+    source_file = SourceFile("UnderTest.py", ".toBe( \n '78' \n )")
+    assert (
+        str(source_file.parse_to_be_like(1).function_call_plus_arg)
+        == ".toBe( \n '78' \n )"
+    )
+    assert str(source_file.parse_to_be_like(1).arg) == "'78'"
+
+    source_file = SourceFile("UnderTest.py", ".toBe('\\'')")
+    assert str(source_file.parse_to_be_like(1).function_call_plus_arg) == ".toBe('\\'')"
+    assert str(source_file.parse_to_be_like(1).arg) == "'\\''"
+
+
+def test_multi_line_string():
+    source_file = SourceFile("UnderTest.py", ".toBe('''7''')")
+    assert (
+        str(source_file.parse_to_be_like(1).function_call_plus_arg) == ".toBe('''7''')"
+    )
+    assert str(source_file.parse_to_be_like(1).arg) == "'''7'''"
+
+    source_file = SourceFile("UnderTest.py", ".toBe('''7''')")
+    assert (
+        str(source_file.parse_to_be_like(1).function_call_plus_arg) == ".toBe('''7''')"
+    )
+    assert str(source_file.parse_to_be_like(1).arg) == "'''7'''"
+
+    # source_file = SourceFile("UnderTest.py", ".toBe('''\n7\n''')")
+    # assert (
+    #     str(source_file.parse_to_be_like(1).function_call_plus_arg)
+    #     == ".toBe('''\n7\n''')"
+    # )
+    # assert str(source_file.parse_to_be_like(1).arg) == "'''\n7\n'''"
+
+    # source_file = SourceFile("UnderTest.py", ".toBe(''' ' '' ' ''')")
+    # assert (
+    #     str(source_file.parse_to_be_like(1).function_call_plus_arg)
+    #     == ".toBe(''' ' '' ' ''')"
+    # )
+    # assert str(source_file.parse_to_be_like(1).arg) == "''' ' '' ' '''"
```

Comment:
Either:

- fix the code so that these tests work
- remove these tests from the PR, and comment in the issue that these parts aren't complete yet
  - if you do this, try to link to the specific Kotlin tests in the issue